### ISSUE TITLE
AutoInsetter - Update contentOffset unexpectedly

### ISSUE THREAD

**Issue by lowip** [uias/Tabman#300]
### New Issue Checklist

- [ x] Updated Tabman to the latest version
- [ x] I took a look at the [Troubleshooting Guide](https://github.com/uias/Tabman/blob/master/Docs/TROUBLESHOOTING.md)
- [ x] I searched for [existing GitHub issues](https://github.com/uias/Tabman/issues)

### Issue Description

I have a `TabmanViewController` with the `currentViewController` having `prefersStatusBarHidden` set to true.
When selecting one of the cells in my `currentViewController`, a modal view controller is presented with `prefersStatusBarHidden` set to false.

This triggers a `viewDidLayoutSubviews() -> setNeedsChildAutoInsetUpdate() -> autoInsetter.inset(...)`

It goes in here: https://git.io/vhiyH
```swift
if scrollView.contentInset != requiredContentInset {
                  
  let isTopInsetChanged = requiredContentInset.top != scrollView.contentInset.top
  
  scrollView.contentInset = requiredContentInset
  scrollView.scrollIndicatorInsets = requiredContentInset
                  
  // only update contentOffset if the top contentInset has updated.
  if isTopInsetChanged {
      var contentOffset = scrollView.contentOffset
      contentOffset.y = -requiredContentInset.top // ⚠️We lose the current content offset here
      scrollView.contentOffset = contentOffset
  }
}
```
As you can see, the `scrollView.contentOffset.y` is updated regardless of the current `contentOffset`.

I propose the following change:
```swift
if scrollView.contentInset != requiredContentInset {
                
  let isTopInsetChanged = requiredContentInset.top != scrollView.contentInset.top
  let topInsetDelta = requiredContentInset.top - scrollView.contentInset.top
  
  scrollView.contentInset = requiredContentInset
  scrollView.scrollIndicatorInsets = requiredContentInset
                  
  // only update contentOffset if the top contentInset has updated.
  if isTopInsetChanged {
      var contentOffset = scrollView.contentOffset
      contentOffset.y -= topInsetDelta
      scrollView.contentOffset = contentOffset
  }
}
```
Which updates the `contentOffset` to the delta of the inset change.

I can prepare a pull request if need be.

### Other useful things

A video of the issue: https://gfycat.com/PleasedIcyBichonfrise

**Comment by msaps** [comment]
@lowip great find! And thanks so much for the detailed report/fix. 

I've opened a PR in AutoInsetter and will try to get a release done ASAP 🥇

**Comment by msaps** [comment]
@lowip released AutoInsetter [1.2.4](https://github.com/uias/AutoInsetter/releases/tag/1.2.4) 😄

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by lowip** [comment]
Great! Thanks for your reactivity, and your suite of libraries :)
